src/writer.rs

```diff
@@ -3,18 +3,19 @@ use crate::{
     UartAddress,
 };
 
-/// Convenience type for interacting unidirectionally (write-only) with a 16550 UART device.
-///
-/// It should be noted that this implementation is meant to be a refernece or testing
-/// implementation, as it is extremely slow, uses blocking IO, and does not utilize the FIFO.
+/// Convenience type for interacting unidirectionally (write-only) with a 16550
+/// UART device. It should be noted that this implementation is meant to be a
+/// refernece or testing implementation, as it is extremely slow, uses blocking
+/// IO, and does not utilize the FIFO.
 pub struct UartWriter<A: UartAddress>(Uart<A, Data>);
 
 impl<A: UartAddress> UartWriter<A> {
     /// ### Safety
     ///
-    /// - `address` must be a valid serial address pointing to a UART 16550 device.
+    /// - `address` must be a valid serial address pointing to a UART 16550
+    ///   device.
     /// - `address` must not be read from or written to by another context.
-    pub unsafe fn new(address: A) -> Option<Self> {
+    pub unsafe fn new(address: A, loopback_test: bool) -> Option<Self> {
```

Comment:
But then the caller would have to call `unwrap`  necessarily if test is turned off.

Reply:
That's a good point. The function won't return `Option<Self>`, instead returning `Self`, and optionally performing the loopback test, then panicking if the test failed.